src/main/java/io/github/darealturtywurty/turtychemistry/core/init/BlockEntityInit.java

```diff
@@ -12,6 +14,12 @@ public final class BlockEntityInit extends AbstractInit {
     public static final DeferredRegister<BlockEntityType<?>> BLOCK_ENTITIES = DeferredRegister
         .create(ForgeRegistries.BLOCK_ENTITY_TYPES, TurtyChemistry.MODID);
 
-    public static final RegistryObject<BlockEntityType<ClayAlloyFurnaceBlockEntity>> CLAY_ALLOY_FURNACE = BLOCK_ENTITIES.register("clay_alloy_furnace",
-            () -> BlockEntityType.Builder.of(ClayAlloyFurnaceBlockEntity::new, BlockInit.CLAY_ALLOY_FURNACE.get()).build(null));
+    public static final RegistryObject<BlockEntityType<ClayAlloyFurnaceBlockEntity>> CLAY_ALLOY_FURNACE = registerNewBlockEntity("clay_allow_furnace",ClayAlloyFurnaceBlockEntity::new,BlockInit.CLAY_ALLOY_FURNACE.get());
+
+
+
+    private static <T extends BlockEntity>RegistryObject<BlockEntityType<T>> registerNewBlockEntity(final String name, final BlockEntityType.BlockEntitySupplier<T> blockEntitySupplier, final Block block)
+    {
+        return BLOCK_ENTITIES.register(name,() -> BlockEntityType.Builder.of(blockEntitySupplier,block).build(null));
+    }
```

Comment:
```suggestion
    private static <T extends BlockEntity> RegistryObject<BlockEntityType<T>> registerNewBlockEntity(final String name, final BlockEntityType.BlockEntitySupplier<T> blockEntitySupplier, final Block... blocks) {
        return BLOCK_ENTITIES.register(name, () -> BlockEntityType.Builder.of(blockEntitySupplier, blocks).build(null));
    }
```

Reply:
why a variable array of blocks?

Reply:
Well because that's what its meant to be.

Reply:
ooh ok